<?xml version="1.0"?>
<launch>
  <!-- Arguments -->
  <arg name="image_topic" default="/camera/image_raw" />
  <arg name="points_topic" default="/velodyne_points" />
  <arg name="model_path" default="$(find lidar_vision_detection)/models/yolov3.weights" />
  <arg name="config_path" default="$(find lidar_vision_detection)/models/yolov3.cfg" />
  <arg name="class_names_file" default="$(find lidar_vision_detection)/models/coco.names" />
  <arg name="confidence_threshold" default="0.5" />
  <arg name="nms_threshold" default="0.4" />
  <arg name="input_width" default="416" />
  <arg name="input_height" default="416" />
  <arg name="camera_frame_id" default="camera" />

  <!-- Vision detection node -->
  <node pkg="lidar_vision_detection" type="vision_detection_node" name="vision_detection_node" output="screen">
    <remap from="image" to="$(arg image_topic)" />
    <remap from="points" to="$(arg points_topic)" />
    <param name="model_path" value="$(arg model_path)" />
    <param name="config_path" value="$(arg config_path)" />
    <param name="class_names_file" value="$(arg class_names_file)" />
    <param name="confidence_threshold" value="$(arg confidence_threshold)" />
    <param name="nms_threshold" value="$(arg nms_threshold)" />
    <param name="input_width" value="$(arg input_width)" />
    <param name="input_height" value="$(arg input_height)" />
    <param name="camera_frame_id" value="$(arg camera_frame_id)" />
  </node>
</launch> 